builder/dispatchers.go

```diff
@@ -339,11 +339,19 @@ func expose(b *Builder, args []string, attributes map[string]bool, original stri
 		b.Config.ExposedPorts = make(nat.PortSet)
 	}
 
-	ports, _, err := nat.ParsePortSpecs(append(portsTab, b.Config.PortSpecs...))
+	ports, bindingMap, err := nat.ParsePortSpecs(append(portsTab, b.Config.PortSpecs...))
 	if err != nil {
 		return err
 	}
 
+	for _, bindings := range bindingMap {
+		if bindings[0].HostIp != "" || bindings[0].HostPort != "" {
+			fmt.Fprintf(b.ErrStream, " ---> The mapping to public ports on your host via:\n"+
+				"      Dockerfile EXPOSE (ip:hostPort:containerPort) has been deprecated.\n"+
+				"      Use -p to publish the ports.\n")
+		}
```

Comment:
can we use one phrase - we're using 2 here `mapping to public ports` and `publish the ports`